### PR TITLE
test: 40% coverage of my-view-advance-request page

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at 7ac1120</samp>

Enabled and added unit tests for the `MyViewAdvanceRequestPage` component in `fyle-mobile-app`. This component displays the details and status of an advance request made by a user.

### <samp>🤖 Generated by Copilot at 7ac1120</samp>

> _`xdescribe` gone_
> _Unit tests for component_
> _Spring of code quality_

### <samp>🤖 Generated by Copilot at 7ac1120</samp>

* Enable the test suite for the `MyViewAdvanceRequestPage` component by replacing `xdescribe` with `describe` ([link](https://github.com/fylein/fyle-mobile-app/pull/2495/files?diff=unified&w=0#diff-cbb0b2e749f4ce2842629c17daa5bf3cc3c0101bd6ab437e816a9607efadb436L2-R156))

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/my-view-advance-request/my-view-advance-request.page.spec.ts [hunk 1]
```diff
@@ -1,26 +1,157 @@
...
-        imports: [IonicModule.forRoot()],
-      }).compileComponents();
+  beforeEach(waitForAsync(() => {
+    const advanceRequestServiceSpy = jasmine.createSpyObj('AdvanceRequestService', [
+      'getAdvanceRequest',
+      'getActions',
+      'getInternalStateAndDisplayName',
+      'getActiveApproversByAdvanceRequestId',
+      'modifyAdvanceRequestCustomFields',
+      'pullBackadvanceRequest',
```
Comment (OmkarJ13) [inline src/app/fyle/my-view-advance-request/my-view-advance-request.page.spec.ts:43]: Let's rename this from `pullBackadvanceRequest` to `pullBackAdvanceRequest`

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/fyle/my-view-advance-request/my-view-advance-request.page.spec.ts [hunk 2]
```diff
@@ -1,26 +1,157 @@
...
+    expenseFieldsService = TestBed.inject(ExpenseFieldsService) as jasmine.SpyObj<ExpenseFieldsService>;
+
+    fixture.detectChanges();
+  }));
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('StatisticTypes(): should return statistic types', () => {
```
Comment (hlkavya0213) [inline src/app/fyle/my-view-advance-request/my-view-advance-request.page.spec.ts:115]: StatisticTypes - > statisticTypes, use camel case

File: src/app/fyle/my-view-advance-request/my-view-advance-request.page.spec.ts [hunk 3]
```diff
@@ -1,26 +1,157 @@
...
+    });
+
+    it('should return the receipt details with type as image and thumbnail as file url if extension is png', () => {
+      spyOn(component, 'getReceiptExtension').and.returnValue('png');
+      const mockFileObject = cloneDeep(advanceRequestFileUrlData2[0]);
+      const result = component.getReceiptDetails(mockFileObject);
+      expect(component.getReceiptExtension).toHaveBeenCalledOnceWith(mockFileObject.name);
+      expect(result).toEqual({
+        type: 'image',
+        thumbnail:
```
Comment (hlkavya0213) [inline src/app/fyle/my-view-advance-request/my-view-advance-request.page.spec.ts:144]: we shouldn't be showing the thumbnail replace with dummy image, pls check latest master

Comment (suyashpatil78) [reply src/app/fyle/my-view-advance-request/my-view-advance-request.page.spec.ts:144]: In the `my-view-advance-request` page, the changes are not live yet.
ref - https://github.com/fylein/fyle-mobile-app/blob/5357414c60758dd3838d5d874c669bc92891dd25/src/app/fyle/my-view-advance-request/my-view-advance-request.page.ts#L89

Comment (suyashpatil78) [reply src/app/fyle/my-view-advance-request/my-view-advance-request.page.spec.ts:144]: We are checking for the expected return value of the function `getReceiptDetails`.

**Review by hlkavya0213 — APPROVED** [review]

**Pull request merged**